Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -218,15 +218,39 @@ WarpX::ComputeFaceExtensions(){
             + std::to_string(N_ext_faces_after_eight_ways(2))
         );
     }
+
+    bool using_bck = false;
+
+    // If any cell could not be extended we use the BCK method to stabilize them
+#if !defined(WARPX_DIM_XZ) && !defined(WARPX_DIM_RZ)
     if (N_ext_faces_after_eight_ways(0) > 0) {
-        amrex::Abort(Utils::TextMsg::Err("Some x faces could not be extended"));
+        ApplyBCKCorrection(0);
+        using_bck = true;
     }
+#endif
+
     if (N_ext_faces_after_eight_ways(1) > 0) {
-        amrex::Abort(Utils::TextMsg::Err("Some y faces could not be extended"));
+        ApplyBCKCorrection(1);
+        using_bck = true;
     }
+
+#if !defined(WARPX_DIM_XZ) && !defined(WARPX_DIM_RZ)
     if (N_ext_faces_after_eight_ways(2) > 0) {
-        amrex::Abort(Utils::TextMsg::Err("Some z faces could not be extended"));
+        ApplyBCKCorrection(2);
+        using_bck = true;
     }
+#endif
```

Comment:
A question here, just to make sure I understand this correctly:
- we may apply the BCK correction along x only in 3D (not in 2D, not in RZ);
- we may apply the BCK correction along y in all geometries (what would this do in 2D or RZ, though?);
- we may apply the BCK correction along z only in 3D (not in 2D, not in RZ).

Did I get this right from the code?

Reply:
Yes, you are right. So the point here is that in RZ and 2D the x- and z- components of the magnetic field are not updated and for this reason we don't need any face extensions and we don't need to do anything in x and z and we can skip the call to `ApplyBCKCorrection`.
In 2D/RZ the BCK correction works in the same way as in 3D which is why I don't do anything special.

I have to say that the ECT+RZ could work in theory but it's not tested (yet) and in a separate PR I could even insert an abort somewhere to avoid this combination (if it's not already forbidden, not sure).

If you have better ideas to handle this I am happy to modify the current implementation.

Reply:
Regarding ECT+RZ: good point. 
In fact, even RZ with stair-case is not yet implemented: https://github.com/ECP-WarpX/WarpX/pull/2707